### DISCUSSION TITLE
Manufacturing orders' quantity completed

### DISCUSSION THREAD

**Discussion by matteocarcano** [frePPLe/frepple#612] [Q&A]
Hi, 
i'd like to have some clarifications regarding the following feature:

1) i want to update the manufacturing orders' quantity completed (MO status=confirmed): so i write 14 in the field "quantity completed" (see image below):

![image](https://github.com/frePPLe/frepple/assets/129165989/eddb7b84-cf2b-49fe-9728-3a04b303dc67)

2) i click "save" button 

3) i obtain MO with 14 table leg completed and 20 yet to do (it's what i expect); therefore, i've 20 wooden beam requirements (see image below):

![image](https://github.com/frePPLe/frepple/assets/129165989/64ee4c8a-68e5-49cf-aaba-3e2d4090c7d1)

4) i launch the plan from execute sheet

5) i obtain the following result, that apparently has no sense (see image below):

- (initial) quantity has been reducted by the quantity completed (34-14= 20)
- quantity completed remains the same (14)
- wooden beam's requirement has decreased to 6 (20-14=6)

![image](https://github.com/frePPLe/frepple/assets/129165989/697ff0e5-8901-4410-8d5f-dd1e7b579faa)

Is this software behavior correct or is it a bug?

I'd like to know how to manage quantity updates on confirmed MO.

Thank you

**Comment by jdetaeye** [comment]
It's a feature: FrePPLe expects that onhand inventory of the item would have increased with completed quantity of 14 - frepple doesn't do that by itself.
And it's a bug: after replanning I wouldn't have expected that the quantity of the complete MO changes.  It should have stayed at 34.

Frepple is a planning tool, not an ERP, MES or shop floor control tool.  The inventories and actual movements in the shop floor need to recorded&tracked in those other applications.    

In summary, the quantity-completed feature is not fully robust production-ready feature for use as a shop floor control tool.

**Comment by matteocarcano** [comment]
Ok i know that Frepple is a planning tool but i think this function is fundamental for production's shceduling.

Do you think this bug can be fixed?

**Reply by jdetaeye** [reply]
Feel free to submit a pull request.

It's not a high priority for the frepple team. So if it's a priority for you,  you'll have to help out.

**Reply by matteocarcano** [reply]
Ok i can try to fix it but I'm not so expert ; can you help me to undestand the code? I've already read documentation but code is very complex. 
Can you help me to find where i have to edit code?
